### PR TITLE
add common functions to test Addon config APIs

### LINKED ISSUE #2388: Create a library in tanzu-framework for addons API testing
Library should provide for the following capabilities:
1. Library to check if package installs are successful
2. Library to check if addon configs are good
3. Ability to customize the Addon Configs

### PR DESCRIPTION
**Pull request by ggpaue**
### Which issue(s) this PR fixes

Fixes #2388 #2456 

### Release note

```release-note
package-based-lcm: add common functions to test addon packages
```

### PR Checklist

- [ ] Squash the commits into one or a small number of logical commits
      
- [ ] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [ ] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

File: pkg/v1/tkg/test/tkgctl/shared/package.go [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+	restConfig, err := clientcmd.RESTConfigFromKubeConfig(rawConfig)
+	Expect(err).ToNot(HaveOccurred(), "Failed to create rest config ")
+
+	client, err := client.New(restConfig, client.Options{Scheme: scheme})
+	Expect(err).ToNot(HaveOccurred(), "Failed to create a cluster client")
+
+	return client, nil
+}
+
+// getClusterBootstrap gets ClusterBootstrap resource with the provided object key
```
Comment (adduarte) [inline pkg/v1/tkg/test/tkgctl/shared/package.go:50]: "provided object key" ?  I think you mean with provided clustername and namespace. no key is provided as a paramenter.  Also probably don't even need comment. Is a private function and is pretty obvious from just the function signature what it does.

File: pkg/v1/tkg/test/tkgctl/shared/package.go [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+	Eventually(func() bool {
+		err := k8sClient.Get(ctx, objKey, clusterBootstrap)
+		return err == nil
+	}, waitTimeout, pollingInterval).Should(BeTrue())
+
+	Expect(clusterBootstrap).ShouldNot(BeNil())
+	return clusterBootstrap
+}
+
+// getPackageInstall get PackageInstall resource with the provided object key
```
Comment (adduarte) [inline pkg/v1/tkg/test/tkgctl/shared/package.go:64]: Same as before, you can do without the comment. Just more text to mantain.

**Comment by ggpaue** [comment]
For loop on AdditionalPackages

**Comment by ggpaue** [comment]
ips conditional on csi and cpi packages

File: pkg/v1/tkg/test/tkgctl/shared/e2e_common.go [hunk 3]
```diff
@@ -137,6 +142,26 @@ func E2ECommonSpec(context context.Context, inputGetter func() E2ECommonSpecInpu
 		})
 		Expect(err).To(BeNil())
 
+		By(fmt.Sprintf("Generating credentials for management cluster %q", input.E2EConfig.ManagementClusterName))
```
Comment (shyaamsn) [inline pkg/v1/tkg/test/tkgctl/shared/e2e_common.go:145]: Run this only for devcc plan

**Comment by ggpaue** [comment]
close this PR since new PR opened: https://github.com/vmware-tanzu/tanzu-framework/pull/2743

**Pull request closed without merging**